src/_pytest/fixtures.py

```diff
@@ -154,12 +153,10 @@ def get_scope_node(node: nodes.Node, scope: Scope) -> nodes.Node | None:
 
 
 def getfixturemarker(obj: object) -> FixtureFunctionMarker | None:
-    """Return fixturemarker or None if it doesn't exist or raised
-    exceptions."""
-    return cast(
-        Optional[FixtureFunctionMarker],
-        safe_getattr(obj, "_pytestfixturefunction", None),
-    )
+    """Return fixturemarker or None if it doesn't exist"""
+    if isinstance(obj, FixtureFunctionDefinition):
```

Comment:
We should try to make fixture definition take the place of the marker in usage

Reply:
I added a todo for this to work on it separately. There are different usages of this function that I don't fully understand and need some time to work on it. Do you think that would work? Otherwise I can work on replacing those usages as well.

Reply:
Let's land this first and figure the other usages as a follow up then

I wasn't aware and we shouldn't let accidential scope creep stop this